examples/lie_labs_api.py

```diff
@@ -0,0 +1,20 @@
+import torch
+
+import theseus.labs.lie as lie
+
+g = lie.rand(lie.SE3, 5)
+x = torch.rand(5, 6)
+
+z = g.retract(x)
+
+try:
+    y = g + x
+except RuntimeError as e:
+    print(e)
+
+y = g + lie.TangentTensor(x)
+
+torch.testing.assert_close(z._t, y._t)
```

Comment:
We could probably override assert close too?

Reply:
I did a quick test and with some slight modification this works if we subclass torch.Tensor (so it works for `TangentTensor`). 

I was playing around with subclassing LieTensor over the weekend, but had some issues with the constructor not being compatible with the superclass. I'll keep looking to see if there is a way to make this work.